src/main/scala/mesosphere/marathon/core/appinfo/impl/AppInfoBaseData.scala

```diff
@@ -188,6 +192,19 @@ class AppInfoBaseData(
       )
     }
 
+  @SuppressWarnings(Array("all")) // async/await
+  def podInstanceStatus(podId: PathId, instance: Instance): Future[Option[PodInstanceStatus]] =
+    async { // linter:ignore UnnecessaryElseBranch
+      val maybePodSpec: Option[PodDefinition] = await(podRepository.getVersion(
```

Comment:
We know all instances (+versions) that we want to fetch. 
Perhaps it makes sense to create a separate:
```scala
lazy val podSpecByVersion: Future[Map[Timestamp, PodDefinition]]
```
which loads only the necessary versions in question?